### PR TITLE
Polyfill shadow dom on login page to help password managers

### PR DESCRIPTION
**Pull request by rianadon**
Continuing off my comment on #3133, I figured I'd submit a PR just to show the changes I'm proposing.

## Proposed change

The login page HTML template (`authorize.html.template`) has an added `<script>` tag which loads the web components polyfill and force enables the polyfill even if the browser supports web components. This eliminates Shadow DOM boundaries only the login page, which helps password managers find the login form.

Chrome is making good progress supporting its internal password manager on pages with shadow dom, but I expect it will be at least a year until all other browsers and password managers fix this issue as well. It should be fairly easy to revert this commit once support is added.

I also upgraded `@webcomponents/webcomponentsjs` which has the polyfill. I don't think it makes any difference, but figured if the polyfill is going to always be used it may as well be the latest version.

## Breaking change

🤞 Hopefully none 🤞. Removing the shadow dom boundaries has the unfortunate side effect of allowing some styles that were supposed to be isolated to one component now affecting the whole page, so I tried to patch these with some css. I tested myself, even with all login modes enabled, but cannot promise anything.

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue:
- This PR is related to issue or discussion: #3133 (This PR doesn't address the accessibility problems)
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.
  ^ I'm not even sure how I would write a test for this fix.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by homeassistant** [comment]
Hi @rianadon, 

It seems you haven't yet signed a CLA. Please do so [here](https://home-assistant.io/developers/cla_sign_start/?pr=home-assistant&#x2F;frontend#9235).

Once you do that we will be able to review and accept this pull request.

Thanks!

File: yarn.lock [hunk 1]
```diff
@@ -3472,11 +3472,16 @@
   resolved "https://registry.yarnpkg.com/@webcomponents/shadycss/-/shadycss-1.9.1.tgz#d769fbadfa504f11b84caeef26701f89070ec49a"
   integrity sha512-IaZOnWOKXHghqk/WfPNDRIgDBi3RsVPY2IFAw6tYiL9UBGvQRy5R6uC+Fk7qTZsReTJ0xh5MTT8yAcb3MUR4mQ==
 
-"@webcomponents/webcomponentsjs@^2.2.10", "@webcomponents/webcomponentsjs@^2.2.7":
+"@webcomponents/webcomponentsjs@^2.2.10":
   version "2.2.10"
   resolved "https://registry.yarnpkg.com/@webcomponents/webcomponentsjs/-/webcomponentsjs-2.2.10.tgz#6f6bee0277833ae98d7e5b46f1e0fdb48cd5ff44"
   integrity sha512-5dzhUhP+h0qMiK0IWb7VNb0OGBoXO3AuI6Qi8t9PoKT50s5L1jv0xnwnLq+cFgPuTB8FLTNP8xIDmyoOsKBy9Q==
 
+"@webcomponents/webcomponentsjs@^2.2.7":
```
Comment (balloob) [inline yarn.lock:3480]: Include a resolution in package.json such that we only install ever a single polyfill.

**Comment by bramkragten** [comment]
I'm a bit worried this will break at some point, and that we then suddenly can't support password managers anymore, getting more issues from users as it would be a regression...

(Side note; I'm surprised that we don't have to/can't use `lit/polyfill-support` since we use `ShadyDOM`... 🤔 Using it actually breaks styling)

**Comment by rianadon** [comment]
Thank you for the changes @bramkragten! I see your point; patching the styles that weren't correctly scoped to their components took time, and if the login page has any future big changes patching the styles again would be a pain.

However, I don't think the login page would need to change much in the future, since it's just a login page. And if it does need a major redesign, then perhaps the code can be rewritten with password manager support in mind so it doesn't even need the polyfill. Or maybe password managers will have support for elements nested within shadow dom trees by then.

I tried using `lit/polyfill-support` on top of my changes but I couldn't notice a different. I wonder what we did differently.

**Comment by rianadon** [comment]
Ah nevermind. I realized I didn't do the build correctly. Now the buttons have lost their styling. Here's what I have:

```diff
diff --git a/build-scripts/gulp/gather-static.js b/build-scripts/gulp/gather-static.js
index 58383c6ed..32a193052 100644
--- a/build-scripts/gulp/gather-static.js
+++ b/build-scripts/gulp/gather-static.js
@@ -59,6 +59,11 @@ function copyPolyfills(staticDir) {
     npmPath("@webcomponents/webcomponentsjs/bundles/webcomponents-sd.js.map"),
     staticPath("polyfills/")
   );
+
+  copyFileDir(
+    npmPath("lit/polyfill-support.js"),
+    staticPath("polyfills/")
+  );
 }
 
 function copyLoaderJS(staticDir) {
diff --git a/src/html/_js_base.html.template b/src/html/_js_base.html.template
index 56f43c6be..98ff1bbaa 100644
--- a/src/html/_js_base.html.template
+++ b/src/html/_js_base.html.template
@@ -21,5 +21,6 @@
         document.write("<script src='/static/polyfills/webcomponents-bundle.js'><"+"/script>");
       } else if (window.ShadyDOM && window.ShadyDOM.force) {
         document.write("<script src='/static/polyfills/webcomponents-sd.js'><"+"/script>");
+        document.write("<script src='/static/polyfills/polyfill-support.js'><"+"/script>");
       }
     </script>
diff --git a/src/html/authorize.html.template b/src/html/authorize.html.template
index 4ce43c7ed..b04d833b1 100644
--- a/src/html/authorize.html.template
+++ b/src/html/authorize.html.template
@@ -44,12 +44,12 @@
     </script>
     <%= renderTemplate('_js_base') %>
     <%= renderTemplate('_preload_roboto') %>
-    <style>
-      /* Polyfill :host styles */
-      ha-icon-next { width: var(--mdc-icon-size, 24px); height: var(--mdc-icon-size, 24px); }
-      /* Polyfill styles that are misapplied since ShadowDOM's isolation no longer exists */
-      p { margin-top: 1em; }
-    </style>
+    <!-- <style> -->
+    <!--   /* Polyfill :host styles */ -->
+    <!--   ha-icon-next { width: var(--mdc-icon-size, 24px); height: var(--mdc-icon-size, 24px); } -->
+    <!--   /* Polyfill styles that are misapplied since ShadowDOM's isolation no longer exists */ -->
+    <!--   p { margin-top: 1em; } -->
+    <!-- </style> -->
 
     <script crossorigin="use-credentials">
       import("<%= latestPageJS %>");
```

**Comment by bramkragten** [comment]
OK, I just imported it in the `authorize.ts` entrypoint, but we get the same result so, doesn't matter...

**Comment by rianadon** [comment]
@bramkragten Are there any changes I could incorporate to make this PR more palatable?

At the current state it seems with the way things are now, all `:host` styles in the `paper-` elements (i.e. the text inputs and lables) in the Polymer library work with just the inclusion of the web components polyfills.

`:host` styles in the home assistant components and material-web-components that use `lit-element` need to be polyfilled ourself (maybe this is what `lit/polyfill-support` is supposed to do but it is a shame that library does not work). But as long as the login page does not include more components from the home assistant or material web components libraries, this is not an issue.

What is an issue is I just found the PR breaks tabbing 🤮 . Awfully weird but I'll look into that.

**Comment by bramkragten** [comment]
I also noticed that the inputs sometimes loose focus when you type the first letter...

**Pull request closed without merging**